shopelectro/selenium/pages/page.py

```diff
@@ -1,9 +1,20 @@
+from functools import wraps
+
 from shopelectro.selenium import SiteDriver
 
 from selenium.webdriver.common.by import By
 from selenium.webdriver.support import expected_conditions as EC
 
 
+def matched_url(fn):
```

Comment:
The responsibility of this decorator isn't to wait for a page to load
 
Right now we have to synchronize the browser location and `Page` abstraction manually to prevent an inconsistent state issue, so i have created this decorator to forbid usage of wrong `Page` instance at a specific browser location
 
If we will make a method instead of the decorator, it maybe will be more clear from semantic point of view, but we still have to manually call it

I want to propose `Navigation` class instead that will responsible for this synchronization.
I will create todo for it and edit the decorator's comment  to clarify its purpose

Reply:
@artemiy312 , i thought about separated class too, but it seems to me too much to create separated class for one method. The url sync mech is good related to Page class.

>  have to synchronize the browser location and Page abstraction manually to prevent an inconsistent state issue

but this sync is just common waiting by it's semantic. Waiting text in input has the same nature, but we use term "wait", not sync.
Anyway `Page.sync` method (or `sync_page` deco) looks much good then `matched_urls`